pkg/storage/util.go

```diff
@@ -45,3 +51,31 @@ func DeduplicateEvents(events []*cadf.Event) []*cadf.Event {
 
 	return result
 }
+
+// indexName generates the index name for a given tenantID.
+// If tenantID is empty, queries use the audit-* wildcard (cross-tenant).
+// When a tenantID is provided, only audit-<tenantID>* is queried.
+func indexName(tenantID string) string {
+	index := "audit-*"
+	if tenantID != "" {
+		index = fmt.Sprintf("audit-%s*", tenantID)
+	}
+	return index
+}
+
+// TruncateHierarchicalAttribute truncates slash-separated attributes to maxDepth levels.
+// This is used for hierarchical attribute values like "service/compute/instance".
+// If maxDepth is 0 or the attribute has no slashes, returns the attribute unchanged.
+// Example: TruncateHierarchicalAttribute("service/compute/instance", 2) returns "service/compute"
+func TruncateHierarchicalAttribute(attribute string, maxDepth int) string {
+	if maxDepth == 0 || !strings.Contains(attribute, "/") {
+		return attribute
+	}
+
+	parts := strings.Split(attribute, "/")
+	if len(parts) <= maxDepth {
+		return attribute
+	}
+
+	return strings.Join(parts[:maxDepth], "/")
+}
```

Comment:
I would rename this function:

```diff
-func TruncateHierarchicalAttribute(attribute string, maxDepth int) string {
+func SplitBySlashN(str string, n int) string {
```

I get the intention to keep the name appropriate for the use-case, but I would argue that it easier to understand the logic by having a literal name (of the actual operations) and instead adding flavour through a comment. Very similar to what you already have:

```

// SplitBySlashN takes a slash-separated string and returns a prefix up to the n-th "/".
// This is used for truncating hierarchical attribute values like "service/compute/instance" to a different depth.
// If n is 0 or the str has no slashes this isa noop.
// Example: SplitBySlashN("service/compute/instance", 2) returns "service/compute".
```